mcproto/types/nbt.py

```diff
@@ -464,14 +469,24 @@ def value(self) -> PayloadType:
         """Get the payload of the NBT tag in a python-friendly format."""
         raise NotImplementedError
 
+    @override
+    def __hash__(self) -> int:
+        return hash((self.name, self.payload, type(self)))
+
+    @override
+    def __eq__(self, value: object) -> bool:
+        if not isinstance(value, NBTag):
+            return NotImplemented
+        return (self.name, self.payload, type(self)) == (value.name, value.payload, type(value))
+
 
 # endregion
 # region NBT tags types
 
 
 @final
-@define
-class EndNBT(NBTag):
+@define(hash=False, eq=False)
+class EndNBT(NBTag, Hashable):
```

Comment:
`NBTag` already inherits from `Hashable`, why is it here again?

- Same question for all occurences